src/v/cluster/health_monitor_backend.cc

```diff
@@ -271,6 +275,155 @@ void health_monitor_backend::abort_current_refresh() {
     }
 }
 
+namespace {
+void record_risk_in_report(
+  node_restart_risk_report& report,
+  model::node_id self,
+  const leader_status& ls,
+  const model::topic_namespace& nt,
+  model::partition_id pid) {
+    partition_risk restart_risk = ls.manifested_risk;
+    for (auto& r : ls.restart_risks) {
+        if (r.node == self) {
+            restart_risk |= r.risk;
+            break;
+        }
+    }
+
+    vlog(
+      clusterlog.trace,
+      "record_risk_in_report nt={}, pid={}, risks={}",
+      nt,
+      pid,
+      restart_risk);
+
+    switch (restart_risk()) {
+    case partition_risk::c::no_risk():
+        return;
+    case partition_risk::c::rf1_offline():
+        report.rf1_offline.push(nt, pid);
+        return;
+    case partition_risk::c::full_acks_produce_unavail():
+        report.full_acks_produce_unavail.push(nt, pid);
+        return;
+    case (
+      partition_risk::c::leaderless
+      | partition_risk::c::full_acks_produce_unavail)():
+        report.leaderless.push(nt, pid);
+        return;
+    case (
+      partition_risk::c::acks1_data_loss
+      | partition_risk::c::full_acks_produce_unavail)():
+        report.acks1_data_loss.push(nt, pid);
+        return;
+    case (
+      partition_risk::c::acks1_data_loss
+      | partition_risk::c::full_acks_produce_unavail
+      | partition_risk::c::leaderless)():
+        report.acks1_data_loss.push(nt, pid);
+        report.leaderless.push(nt, pid);
+        return;
+    default:
+        vassert(
+          false,
+          "Unexpected partition restart risk combination {} for ntp {}",
+          restart_risk,
+          model::ntp{nt.ns, nt.tp, pid});
+    }
+}
+} // namespace
+
+ss::future<result<node_restart_risk_report>>
+health_monitor_backend::get_current_node_restart_risks(
+  model::timeout_clock::time_point deadline) {
+    auto holder = _gate.hold();
+    if (
+      auto ec = co_await maybe_refresh_cluster_health(
+        force_refresh::no, deadline)) {
+        co_return ec;
+    }
+    if (!_restart_risks_collected) {
+        // technically it may be already enabled at the moment,
+        // but was disabled on last collection
+        co_return make_error_code(errc::feature_disabled);
+    }
+
+    chunked_hash_map<
+      model::topic_namespace,
+      chunked_hash_set<model::partition_id>>
+      unclaimed_partitions;
+    node_restart_risk_report report;
+
+    // local report
+    auto local_report_it = std::as_const(_reports).find(_self);
+    if (local_report_it == _reports.cend()) {
+        vlog(
+          clusterlog.debug,
+          "current node is not part of the cluster => no risks");
+        co_return report;
+    };
+    for (const auto& [nt, partitions] : local_report_it->second->topics) {
```

Comment:
we should copy report ptrs and iterate with yields

Reply:
@bashtanov this is still not fixed, right?

Reply:
tip look at the counter version

Reply:
oops, must have been lost when rebasing, will fix